### PR TITLE
Replace blank "type: ignore" with more specific waivers

### PR DESCRIPTION
**Pull request by happz**
Using mypy's error codes, waivers can target the actual erorr, leaving
space for other kind of violations to not be suppressed.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -60,3 +60,20 @@ repos:
     hooks:
       - id: yamllint
         files: ^tmt/schemas/.*\.yaml
+
+  # Yet another static analysis, but these hooks "understand" Python syntax and semantics, and
```
Comment (thrix) [inline .pre-commit-config.yaml:64]: I was reading this comment and I am not sure if it is very helpful :( 

I would just mention some additional hardening on # noqa and # ignore usage?

Comment (happz) [reply .pre-commit-config.yaml]: Yeah, it's misleading. Rewritten & rebased.

**Review by thrix — APPROVED** [review]
LGTM

**Review by FrNecas — APPROVED** [review]
Also LGTM but needs rebase to main

**Comment by happz** [comment]
Rebased.

**Review by psss — APPROVED** [review]
Thanks for improving this!

**Pull request merged**